nibabel/nicom/dicomwrappers.py

```diff
@@ -471,13 +494,28 @@ def image_shape(self):
         frame_indices = np.array(
             [frame.FrameContentSequence[0].DimensionIndexValues
              for frame in self.frames])
-        n_dim = frame_indices.shape[1] + 1
-        # Check there is only one multiframe stack index
-        if np.any(np.diff(frame_indices[:, 0])):
-            raise WrapperError("File contains more than one StackID. Cannot "
-                               "handle multi-stack files")
+        # Check that there is only one multiframe stack index
+        stack_ids = set(frame.FrameContentSequence[0].StackID
+                        for frame in self.frames)
+        if len(stack_ids) > 1:
+            raise WrapperError("File contains more than one StackID. "
+                               "Cannot handle multi-stack files")
+        # Determine if one of the Dimension indices refers to the stack id
+        dim_seq = [(dim.DimensionIndexPointer, dim.FunctionalGroupPointer)
+                   for dim in self.get('DimensionIndexSequence')]
+        # the pointer pair (StackID Dicom-tag, FrameContentSequence Dicom-tag)
+        # that indicates that the corresponding axis in the DimensionIndexValues
+        # attribute refers to the StackID
+        stack_id_dim_pointer = ((0x20, 0x9056), (0x20, 0x9111))
```

Comment:
Thanks for the feedback.    It would be excellent to have a way of returning an attribute name per axis.   I'm hoping to get some time in the next few months to add named axes to nibabel image objects, and it would be very helpful to be able to get names from dicom image axes.